Design/spectrum_number_and_workspace_index_abstraction.md

```diff
@@ -0,0 +1,269 @@
+# Indexing Abstraction Layer
```

Comment:
I very much agree that not having workspace/spectra index in any UI is the ideal.
A point to consider though is that there can be workspaces with rows that are not spectra, for example the outputs from Fit or other algorithms which are just matrices with rows of data.